src/components/AppSessionStateContext.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import {
+  AppSessionStateMonitor,
+  AppSessionState,
+  defaultAppSessionState,
+} from "../abstractions/app-session";
+import { AppSessionStateStatus } from "../abstractions/app-session/app-session-state";
+
+const AppSessionStateContext = createContext<AppSessionState>(
+  defaultAppSessionState
+);
+const AppSessionStateStatusContext = createContext<string>(
+  defaultAppSessionState.status
+);
+
+export const AppSessionStateProvider = ({
+  children,
+  appSessionStateMonitor,
+}: {
+  children: React.ReactNode;
+  appSessionStateMonitor: AppSessionStateMonitor;
+}) => {
+  const [appSessionState, setAppSessionState] = useState(
+    defaultAppSessionState
+  );
+  const [appSessionStateStatus, setAppSessionStateStatus] = useState(
+    defaultAppSessionState.status
+  );
+
+  useEffect(() => {
+    appSessionStateMonitor.onSessionUpdate((newValue) => {
+      setAppSessionState(newValue);
+      setAppSessionStateStatus(newValue.status);
+    });
+  }, []);
+
+  return (
+    <AppSessionStateContext.Provider value={appSessionState}>
+      <AppSessionStateStatusContext.Provider value={appSessionStateStatus}>
```

Comment:
@jcamposmk do you remember that there are some weird effects in the webapp when the session is updated?

I am not sure, but I think that this approach could help.

Many times we do not need all the session data, the status is enough, so many unnecessary renderings could be avoided.